templates/commands/render_action.go

```diff
@@ -78,3 +79,35 @@ func walkAndModify(pos *model.ConfigPos, rfs renderFS, scratchDir, relPath strin
 		return nil
 	})
 }
+
+func templateAndCompileRegexes(regexes []model.String, inputs map[string]string) ([]*regexp.Regexp, error) {
+	compiled := make([]*regexp.Regexp, len(regexes))
+	for i, re := range regexes {
+		templated, err := parseAndExecuteGoTmpl(re.Pos, re.Val, inputs)
+		if err != nil {
+			return nil, err
+		}
+
+		compiled[i], err = regexp.Compile(templated)
+		if err != nil {
+			return nil, model.ErrWithPos(re.Pos, "failed compiling regex: %w", err) //nolint:wrapcheck
+		}
+	}
+
+	return compiled, nil
+}
+
+// Returns a slice where the order of the elements is reverse. Returns the same
+// slice as the input in the case where the length of the input is 0 or 1,
+// because the reverse operation is a no-op on these.
+func reverse[T any](in []T) []T {
```

Comment:
Why would we do this vs iterating over the slice in reverse? It feels unnecessarily expensive.

Reply:
Because `reverse()` seemed more readable me. It lets us use `for _ oneMatch := range allMatches` and avoids creating more index variables. This code already has a confusing number of indices (FindAllSubmatchIndex returns a slice-of-slices-of-indices 😱 ), so avoiding more indices seemed desirable.

 I take it you disagree though; so I switched over to iterating in reverse. I don't have strong feelings on it.